src/ceph-volume/ceph_volume/devices/lvm/zap.py

```diff
@@ -164,6 +240,17 @@ def zap(self):
 
         terminal.success("Zapping successful for: %s" % ", ".join([str(d) for d in self.args.devices]))
 
+    @decorators.needs_root
+    def zap_osd(self):
+        if self.args.osd_id:
+            osd_is_running = systemctl.osd_is_active(self.args.osd_id)
```

Comment:
This won't work for containers. So we need to find a way to skip this check if ceph-volume is invoked from within a container. They are several ways to do it, a straightforward one is to look for the presence of `/.dockerenv`, it's not perfect because it's tight to docker for sufficient for now.

Reply:
This check will be a noop on a container. It will not find anything running for `ceph-osd@X`

How do you see this failing?

Reply:
Sounds good.